rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java

```diff
@@ -849,9 +884,26 @@ public J visitLambdaExpression(LambdaExpressionTree node, Space fmt) {
 
     @Override
     public J visitLiteral(LiteralTree node, Space fmt) {
-        cursor(endPos(node));
+        int endPos = endPos(node);
         Object value = node.getValue();
-        String valueSource = source.substring(((JCLiteral) node).getStartPosition(), endPos(node));
+
+        if (endPos == Position.NOPOS) {
+            if (typeMapping.primitive(((JCLiteral) node).typetag) == JavaType.Primitive.String) {
+                int quote = source.substring(cursor).startsWith("\"\"\"") ? 3 : 1;
+                if (quote == 3) {
+                    endPos = cursor + quote + source.indexOf("\"\"\"", cursor + quote) + quote;
+                } else {
+                    endPos = cursor + quote + value.toString().length() + quote;
+                }
+            } else {
+                endPos = cursor + indexOf(source.substring(cursor),
+                        ch -> Character.isWhitespace(ch) || ",;)]}+-*/%=!<>&|^?:.".indexOf(ch) != -1
+                );
+            }
+        }
+
+        cursor(endPos);
+        String valueSource = source.substring(((JCLiteral) node).getStartPosition(), endPos);
```

Comment:
The reason this had to change at all is because the endpos table is determined in this step https://github.com/openrewrite/rewrite/blob/04628ec7de2429a6d61cc909667aec418d2378fd/rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17Parser.java#L224
and the removal of unapplicable annotations is done here https://github.com/openrewrite/rewrite/blob/04628ec7de2429a6d61cc909667aec418d2378fd/rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17Parser.java#L236 without updating that endpos table, so we can no longer reliable get that information